lib/results.go

```diff
@@ -32,13 +34,22 @@ type Result struct {
 	Body      []byte        `json:"body"`
 	Method    string        `json:"method"`
 	URL       string        `json:"url"`
+	Header    http.Header   `json:"header"`
 }
 
 // End returns the time at which a Result ended.
 func (r *Result) End() time.Time { return r.Timestamp.Add(r.Latency) }
 
 // Equal returns true if the given Result is equal to the receiver.
 func (r Result) Equal(other Result) bool {
+	headerIsEqual := true
+	for headerKey := range r.Header {
+		if r.Header.Get(headerKey) != other.Header.Get(headerKey) {
```

Comment:
This only gets the last value for a given Header key. Fix that and extract this code into a function we  can call below in the return statement.

```go
func headersEqual(a, b http.Header) bool {
    for k, as := range a {
        bs := b[k]
        if len(bs) != len(as) {
            return false
        }
        for i := range as {
            if as[i] != bs[i] {
                return false
            }
        }
    }
    return true
}
```